src/core/Akka.Remote/RemoteSystemDaemon.cs

```diff
@@ -183,6 +278,37 @@ public override IActorRef GetChild(IEnumerable<string> name)
             }
             return ActorRefs.Nobody;
         }
+
+        private void AddChildParentNeedsWatch(IActorRef parent, IActorRef child)
```

Comment:
We need some eyes on this. I think this part is threadsafe but it should be reviewed.